File: backend/src/models/order.ts
import mongoose from "mongoose";

const orderSchema = new mongoose.Schema({
    table: { type: mongoose.Schema.Types.ObjectId, ref: 'Table' },
    meals: [{
        meal: { type: mongoose.Schema.Types.ObjectId, ref: 'Meal' },
        quantity: { type: Number, required: true }
    }],
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: false },
    date: Date,
    status: {
        type: String,
        enum: ['PENDING', 'IN_PROGRESS', 'SERVED', 'PAID', 'CANCELLED','ARCHIVED'],
        default: 'PENDING'
    },
    paymentStatus: {
        type: String,
        enum: ['AWAITING_PAYMENT', 'PAID', 'EXPIRED', 'REFUNDED', 'FAILED'],
        default: 'AWAITING_PAYMENT'
    },
    paymentIntentId: { type: String, required: false },
    paymentIdentifier: { type: String, required: false },
    reservation: { type: mongoose.Schema.Types.ObjectId, ref: 'Reservation', default: null },
    createdAt: { type: Date, default: Date.now },
    notes: String
});

export default mongoose.model('Order', orderSchema);

File: mobile/employe/lib/models/order.dart
class Order {
  String? id;
  String? table;
  List<String>? meals;
  String? user;
  DateTime? date;
  String? status;
  String? paymentStatus;
  String? reservation;
  DateTime? createdAt;
  String? notes;

  Order({
    this.id,
    this.table,
    this.meals,
    this.user,
    this.date,
    this.status,
    this.paymentStatus,
    this.reservation,
    this.createdAt,
    this.notes,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['_id'],
      table: json['table'],
      meals: List<String>.from(json['meals'] ?? []),
      user: json['user'],
      date: json['date'] != null ? DateTime.parse(json['date']) : null,
      status: json['status'],
      paymentStatus: json['paymentStatus'],
      reservation: json['reservation'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      notes: json['notes'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'table': table,
      'meals': meals,
      'user': user,
      'date': date?.toIso8601String(),
      'status': status,
      'paymentStatus': paymentStatus,
      'reservation': reservation,
      'createdAt': createdAt?.toIso8601String(),
      'notes': notes,
    };
  }
}